### PR TITLE
Refactor specialization to run after distribution loops are removed

### PR DESCRIPTION
**Pull request by ThomasRaoux**
This will allow us to remove distribution loops without any interaction with this pass. This also remove dependency to the brittle tiled loop analysis.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/WorkgroupSpecializationPass.cpp [hunk 1]
```diff
@@ -117,101 +68,56 @@ static SmallVector<AffineMinOp> getBoundedTileSizeOps(scf::ForOp forOp,
...
-      // TODO: handle dynamic cases by adding more code to check the
-      // conditions.
-      return;
-    }
-    int64_t ub = ubOp.value();
-    int64_t tileSize = tileSizes[i];
-    if (ub % tileSize != 0) {
-      isAlreadyMultiple = false;
-    }
-    if (ub < tileSize) {
```
Comment (okkwon) [inline compiler/src/iree/compiler/Codegen/Common/WorkgroupSpecializationPass.cpp:173]: Don't we need these checks in the new code too? Especially, the new code covers the dynamic cases, but I don't think the dynamic cases are tested well.

Comment (ThomasRaoux) [reply compiler/src/iree/compiler/Codegen/Common/WorkgroupSpecializationPass.cpp:173]: I thought there would be no affine_min in this case. Let me double check.

Comment (ThomasRaoux) [reply compiler/src/iree/compiler/Codegen/Common/WorkgroupSpecializationPass.cpp:173]: checking again, the tileAndDistribute won't create affine_min in this case so nothing happen

**Review by okkwon — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**